packages/frontend/src/features/sqlRunner/components/Header/HeaderEdit.tsx

```diff
@@ -41,6 +43,37 @@ export const HeaderEdit: FC = () => {
     const config = useAppSelector((state) =>
         selectChartConfigByKind(state, selectedChartType),
     );
+
+    // Store initial chart config to detect if there are any changes later on
+    const [initialSavedSqlChart, setInitialSavedSqlChart] = useState<
```

Comment:
if you can think of an easier way check this, between 2 different redux states, let me know.

Reply:
This is basically the way IMO. 
I'll provide a few ways to simplify this now

Reply:
BTW, an update to this: you could use `createSelector` to check these changes, but since it's isolated to `edit mode` (for now) I believe this is great, and it's already powered by redux selectors